### PR TITLE
query(expressions): add try_cast function

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

1.  add try_cast function 
2.  fix `nullable(const) case`

## Changelog

- New Feature
- Improvement
 

## Related Issues

Fixes #issue

### CODE REVIEW

File: common/planners/src/plan_expression.rs [hunk 1]
```diff
@@ -118,8 +118,6 @@ pub enum Expression {
         expr: Box<Expression>,
         /// The `DataType` the expression will yield
         data_type: DataTypePtr,
-        /// default false, if it's try_cast, is_null is true
```
Comment (b41sh) [inline common/planners/src/plan_expression.rs:121]: typo `is_nullable`

Comment (sundy-li) [reply common/planners/src/plan_expression.rs:121]: It's deletions.

Comment (b41sh) [reply common/planners/src/plan_expression.rs:121]: ok

File: common/datavalues/src/columns/nullable/mod.rs [hunk 2]
```diff
@@ -30,7 +30,32 @@ pub struct NullableColumn {
...
+        }
+
+        if !column.is_const() {
+            Self::new_from_opt(column, validity).arc()
+        } else {
+            let c: &ConstColumn = Series::check_get(&column).unwrap();
+            let inner = c.inner().clone();
+            let validity = validity.map(|b| b.slice(0, 1));
+            let nullable_column = Self::new_from_opt(inner, validity).arc();
+            ConstColumn::new(nullable_column, c.len()).arc()
```
Comment (b41sh) [inline common/datavalues/src/columns/nullable/mod.rs:45]: How about to add some comments to explain this case?

File: tests/suites/0_stateless/02_function/02_0002_function_try_cast.sql [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+SELECT try_cast('-33aa' as signed) is null;
+SELECT try_cast('33 aa' as unsigned) is null;
+SELECT try_cast('-33' as unsigned) is null;
+SELECT try_cast('aa' as unsigned) is null;
+SELECT try_cast('aa' as Float64) is null;
+
+SELECT try_cast(parse_json('null') as float64) is null;
+
+-- TODO(b41sh)
+-- SELECT try_cast(parse_json('"test"') as int32) is null;
```
Comment (b41sh) [inline tests/suites/0_stateless/02_function/02_0002_function_try_cast.sql:19]: as [#4793] has merged, we can test this.

**Pull request merged**